src/blocks/sandbox-image/imports/index.mjs

```diff
@@ -0,0 +1,324 @@
+/**
+ * BU's take on the Image component.
+ */
+
+// External dependencies.
+import classnames from 'classnames';
+import { useState } from '@wordpress/element';
+
+import {
+	MediaPlaceholder,
+	InspectorControls,
+	MediaUploadCheck,
+	MediaUpload,
+} from '@wordpress/block-editor';
+
+import {
+	Button,
+	IconButton,
+	FocalPointPicker,
+	PanelBody,
+	PanelRow,
+	Placeholder,
+} from '@wordpress/components';
+
+import { __ } from '@wordpress/i18n';
+
+// BU dependencies.
+import {
+	useMedia,
+	getImageData,
+	LoadingSpinner,
+} from '@bostonuniversity/block-imports';
+
+/**
+ * Returns the class list for the component based on the current settings.
+ *
+ * @param {string} className Additional classes assigned to the component.
+ */
+const getClasses = ( className ) =>
+	classnames( 'bu-components-image', {
+		[ className ]: className,
+	} );
+
+/**
+ * Export component.
+ *
+ * @param  props
+ * @return {Element} Element to render, in this case an DIV.
+ */
+export const Image = ( props ) => {
+	const {
+		// Setup
+		canEditImage = true,
+		debug = true,
+		tag = 'img',
+		size = 'thumbnail',
+		sourceSizes = [],
+		altSource = 'alt',
+		mediaId = undefined,
+		className = undefined,
+		canEditFocalPoint = true,
+		focalPoint = { x: 0.5, y: 0.5 },
+		// Placeholder
+		label = 'Placeholder label...',
+		instructions = 'Placeholder instructions...',
+		// MediaPlaceholder
+		labels = {
+			title: 'MediaPlaceholder label...',
+			instructions: 'MediaPlaceholder instructions...',
+		},
+		//
+		...rest
+	} = props;
+
+	/**
+	 * States.
+	 */
+	const [ currentFocalPoint, setFocalPoint ] = useState( focalPoint );
+	const [ currentMediaId, setMediaId ] = useState( mediaId );
```

Comment:
I think we can get rid of useState here for `currentMediaId`. Instead we should be able to just make use of the `onSelect` or `onSelectMedia` function passed from the block. 

When the image is picked in the component, `onSelect()` would be called in the block and passed the media object. The block's function for handling that change event would then be responsible for getting the id from the media object and calling setAttribute( mediaId: newMedia.id ); or similar and setting/updating the mediaId attribute. Because the mediaId attribute is passed to the image component it would rerender and use the new mediaId for everything (focal point picker, etc).